### PR TITLE
Fix bug were max fee was not respected for deploy

### PR DESCRIPTION
**Pull request by ksew1**
**Stack**:
- #2310
- #2309
- #2308
- #2307 ⬅

⚠️ *Part of a stack created by [spr](https://github.com/ejoffe/spr). Do not merge manually using the UI - doing so may have unexpected results.*

### CODE REVIEW

**Review by Arcticae — COMMENTED** [review]
Can you please provide a description of what the bug was and how you fixed it? I find it hard to deduce from the code itself

File: crates/sncast/src/helpers/fee.rs [hunk 1]
```diff
@@ -118,288 +130,24 @@ pub enum FeeToken {
 }
 
 #[derive(Debug, PartialEq, CairoDeserialize)]
-pub struct EthFeeSettings {
-    pub max_fee: Option<FieldElement>,
+pub enum ScriptFeeSettings {
```
[the remark this answers is not in the document]
Comment (ksew1) [reply crates/sncast/src/helpers/fee.rs:133]: This struct is only used in `sncast script`. `FeeArgs` is not ergonomic to work with, and `FeeSettings` doesn't have an option to specify `max_fee` as it is used after the `max_fee` calculation. Therefore, I created `ScriptFeeSettings`. However, I am open to different naming suggestions. What do you propose?

File: crates/sncast/tests/integration/fee.rs [hunk 2]
```diff
@@ -21,100 +21,285 @@ async fn get_factory() -> OpenZeppelinAccountFactory<LocalWallet, JsonRpcClient<
 }
 
 #[tokio::test]
-async fn test_eth_fee_get_or_estimate() {
+async fn test_happy_case_eth() {
```
[the remark this answers is not in the document]
Comment (ksew1) [reply crates/sncast/tests/integration/fee.rs:24]: I can't be as it needs rpc provider now

**Comment by ksew1** [comment]
## Bug description

The `max_fee` calculation only occurs when the method `get_or_estimate` is called. However, the `account deploy` functionality does not call this method. 

## Fix

`starknet-rs` automatically estimates the fee. Therefore, we only need to convert `max_fee` to `max_gas` and `max_gas_unit_price`. This calculation is now performed in the conversion method. Consequently, we can delete the `get_or_estimate` method.

File: crates/sncast/src/helpers/fee.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (Arcticae) [reply crates/sncast/src/helpers/fee.rs:133]: I would leave a comment here about the purpose, the name is fine if it's only for script

**Review by Arcticae — APPROVED** [review]

**Comment by ksew1** [comment]
✓ Commit merged in pull request [#2310](https://github.com/foundry-rs/starknet-foundry/pull/2310)

**Pull request closed without merging**